### ISSUE TITLE
Add other languages

### ISSUE THREAD

**Issue by HuseynHuseyni** [codeque-co/codeque#7]
Hi please add other languages like C,C++,Dart,Python

**Comment by jayu** [comment]
Definitely CodeQue will support more languages in future. It's not that straight forward to add support for new languages, but I will work on that in future :)

**Comment by HuseynHuseyni** [comment]
Thank you very much.
I was very happy to hear this.

**Comment by jayu** [comment]
I will actually reopen that issue, maybe others would be eager to comment or list languages that should be supported

**Comment by alessandro-newzoo** [comment]
php, html and html-based languages would be fantastic and super useful! <3

**Comment by shayandavoodii** [comment]
Supporting Python and Julia would be so helpful for me.

**Comment by ChristopherHaws** [comment]
C# and Razor would be fantastic :)

I think before that though, the main config markup languages would probably be the most helpful (yaml, xml, etc)

**Comment by jayu** [comment]
I'm working on PoC support for HTML and CSS, it looks promising :D

**Comment by ELLIOTTCABLE** [comment]
Ours would be OCaml / ReasonML / Melange, but I suspect that's going to be low priority. :P

Looks super-cool, wish I could use it outside plain-TypeScript projects in 2023. )=

**Comment by carlosdelivery** [comment]
Please add ruby support!

**Comment by jayu** [comment]
Hello everyone!

I'm happy to announce that in recent months I worked integrating several new parsers for CodeQue.

CodeQue core is now parser agnostic and basically any parser producing AST in JSON format can be integrated with the tool.

I've also invested time into deep dive in tree-sitter API and as a result I've build foundations for integrating any tree-sitter parser via WebAssembly. 

Tree sitter parser is surprisingly slower comparing to Babel, but after some strange optimizations on JS side that I had to make, it's fast enough to serve for CodeQue search.

Big thanks for @ELLIOTTCABLE and @sschneider-ihre-pvs  for suggesting using tree-sitter.

In recent months I've released support for: 
- HTML
- CSS 
- Python
- Lua

It's worth mentioning that adding basic Lua support via tree sitter took ~4h, which is relatively small amount of time.

I will now close this issue and open a new one for each programming language that was mentioned in this thread.
